### PR TITLE
Add atlassian software packages and modules (confluence, crowd, jira)

### PR DESCRIPTION
**Pull request by globin**
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by puffnfresh** [comment]
Awesome :fireworks:

**Comment by copumpkin** [comment]
This looks awesome! How does one add the license key to these packages? Or must that be done manually after installation?

**Comment by globin** [comment]
Has to be added when configuring after installation. Afaik this cannot be automated.

File: nixos/modules/services/web-apps/atlassian/confluence.nix [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      requires = [ "postgresql.service" ];
+      after = [ "postgresql.service" ];
+
+      path = [ cfg.jrePackage ];
+
+      environment = {
+        CONF_USER = cfg.user;
+        JAVA_HOME = "${cfg.jrePackage}";
+        CATALINA_OPTS = concatStringsSep " " (cfg.catalinaOptions ++ [
+          #"-Djavax.net.ssl.trustStore=${cfg.home}/cacerts"
```
Comment (benley) [inline nixos/modules/services/web-apps/atlassian/confluence.nix]: Is this parameter something that users may actually want to set?

**Review by benley — APPROVED** [review]

File: nixos/modules/services/web-apps/atlassian/confluence.nix (hunk 1 above)
Comment (fpletz) [inline nixos/modules/services/web-apps/atlassian/confluence.nix]: Thanks. Just some leftovers from testing when Oracle JDK didn't have the DST Root CA for Let's Encrypt in their trust store. Will remove before merging.

**Pull request merged**